core/store.ts

```diff
@@ -0,0 +1,16 @@
+import { createStore, combineReducers } from "redux";
+import templatesReducer, {
+  TemplatesState,
+} from "@/resources/utils/templates-store/reducer";
+
+export interface RootState {
+  templates: TemplatesState;
+}
+
+const rootReducer = combineReducers({
+  templates: templatesReducer,
```

Comment:
We can not do this here, we are coupling it, we need to subscribe to the GlobalManager and load it dynamically